JavaScript/Web/AnimationFrame.hs

```diff
@@ -34,22 +34,24 @@ newtype AnimationFrameHandle = AnimationFrameHandle JSRef
 {- |
      Wait for an animation frame callback to continue running the current
      thread. Use 'GHCJS.Concurrent.synchronously' if the thread should
-     not be preempted.
+     not be preempted. This will return the high-performance clock time
+     stamp once an animation frame is reached.
  -}
-waitForAnimationFrame :: IO ()
+waitForAnimationFrame :: IO Double
 waitForAnimationFrame = do
   h <- js_makeAnimationFrameHandle
   js_waitForAnimationFrame h `onException` js_cancelAnimationFrame h
 
 {- |
      Run the action in an animationframe callback. The action runs in a
-     synchronous thread.
+     synchronous thread, and is passed the high-performance clock time
+     stamp for that frame.
  -}
-inAnimationFrame :: OnBlocked -- ^ what to do when encountering a blocking call
-                 -> IO ()     -- ^ the action to run
+inAnimationFrame :: OnBlocked       -- ^ what to do when encountering a blocking call
+                 -> (Double -> IO ())  -- ^ the action to run
                  -> IO AnimationFrameHandle
 inAnimationFrame onBlocked x = do
-  cb <- syncCallback onBlocked x
+  cb <- syncCallback1 onBlocked (x . pFromJSRef)
```

Comment:
Is the `pFromJSRef` the right function for `JSRef -> Double`? I was previously using `pToJSRef` below, but it seems that is now just `jsref`.